DB/src/main/java/io/deephaven/db/v2/utils/sortedranges/SortedRanges.java

```diff
@@ -875,10 +881,15 @@ public boolean advance(final long v) {
                 rangeCurr = Math.min(v, rangeCurr);
                 return true;
             }
-            if (nextRangeIdx < 0 || sar == null) {
+            if (sar == null) {
```

Comment:
Same question as above. Also, did this iterator have "the" bug, or just a resource leak?

Reply:
Not the bug.